pkg/controller/registry/reconciler/reconciler.go

```diff
@@ -195,19 +191,26 @@ func Pod(source *operatorsv1alpha1.CatalogSource, name string, image string, saN
 			NodeSelector: map[string]string{
 				"kubernetes.io/os": "linux",
 			},
-			SecurityContext: &corev1.PodSecurityContext{
-				SeccompProfile: &corev1.SeccompProfile{
-					Type: corev1.SeccompProfileTypeRuntimeDefault,
-				},
-			},
 			ServiceAccountName: saName,
 		},
 	}
 
-	if runAsUser > 0 {
-		pod.Spec.SecurityContext.RunAsUser = &runAsUser
-		pod.Spec.SecurityContext.RunAsNonRoot = pointer.Bool(true)
+	if source.Spec.GrpcPodConfig == nil || source.Spec.GrpcPodConfig.SecurityContextConfig != operatorsv1alpha1.Legacy {
```

Comment:
this should be "if config==nil || config==restricted"

you don't want to treat unknown values as if they were "restricted", you want to treat unknown values as an error.

Reply:
(so you also need an else here to handle unknown values.  Yes validation should prevent it, but if someone adds a new valid enum and doesn't fix up this logic, you want to catch it.)

Reply:
so spec.GrpcPodConfig.SecurityContextConfig is marked as `optional`, which means most of the CatalogSources created will not have the "GrpcPodConfig" struct populated, so config==restricted will actually not be true. 

For example, this is the yaml of the operatorhub catalog applied to the cluster on local installation in Kind: 

```
$ k get catsrc -n olm -o yaml 
apiVersion: v1
items:
- apiVersion: operators.coreos.com/v1alpha1
  kind: CatalogSource
  metadata:
    creationTimestamp: "2022-09-14T17:11:29Z"
    generation: 1
    name: operatorhubio-catalog
    namespace: olm
    resourceVersion: "757"
    uid: 54db9c12-be5e-4b0b-9e16-3d1d80acaf54
  spec:
    displayName: Community Operators
    image: quay.io/operatorhubio/catalog:latest
    publisher: OperatorHub.io
    sourceType: grpc
    updateStrategy:
      registryPoll:
        interval: 60m
  status:
    connectionState:
      address: operatorhubio-catalog.olm.svc:50051
      lastConnect: "2022-09-14T17:11:59Z"
      lastObservedState: READY
    registryService:
      createdAt: "2022-09-14T17:11:29Z"
      port: "50051"
      protocol: grpc
      serviceName: operatorhubio-catalog
      serviceNamespace: olm
kind: List
metadata:
  resourceVersion: ""
  selfLink: ""
```

I'm guessing that means when a third value is introduced, grpcpodConfig.SecurityContextConfig should be marked as `required` and then this can be changed to explicitly check config==restricted. 

Coz the other option is "config==restricted" || config==nil", which doesn't sound that explicit anyway. Looks like even with defining a default value for it, if the top level grpcPodConfig isn't defined at all, the entire struct remains nil. 

I've added a [comment](https://github.com/operator-framework/operator-lifecycle-manager/pull/2845/files#diff-fffdeef1fc140a5dc5dc92dda323f567a6e46fc2ecbb0b91ba907acd02bde50dR198-R204) to prevent someone accidentally introducing faulty logic when a third enum is added